src/main/java/nl/tudelft/contextproject/model/level/MSTBasedLevelFactory.java

```diff
@@ -0,0 +1,340 @@
+	 */
+	protected ArrayList<Stack<Vec2I>> getCorridorLocations() {
+		ArrayList<Stack<Vec2I>> corridorList = new ArrayList<Stack<Vec2I>>();
+		CorridorEdge currentCorridor;
+		for (Integer corridorID: chosenEdges) {
+			currentCorridor = edges.get(corridorID);
+			corridorList.add(CorridorBreadthFirstSearch.breadthFirstSearch(mazeTiles, currentCorridor.start.location, currentCorridor.end.location));
+		}
+		return corridorList;
+	}
+
+	/**
+	 * This method adds walls to corridors on the map.
+	 * Checks if a Tile is a corridor, if true add a wall to all Null TileTypes.
+	 *
+	 * @param map
+	 *		the map in which to place the corridor walls
+	 */
+	protected static void carveCorridorWalls(MazeTile[][] map) {
+		int width = map.length;
+		int heigth = map[0].length;
+		for (int i = 0; i < width; i++) {
+			for (int j = 0; j < heigth; j++) {
+				if (map[i][j] != null && map[i][j].getTileType() == TileType.CORRIDOR) {
+					//Check North
+					if (j != 0 && map[i][j - 1] == null) {
+						map[i][j - 1] = new MazeTile(i, j - 1, TileType.WALL);
+					}
+
+					//Check South
+					if (j != heigth - 1 && map[i][j + 1] == null) {
+						map[i][j + 1] = new MazeTile(i, j + 1, TileType.WALL);
+					}
+
+					//Check West
+					if (i != 0 && map[i - 1][j] == null) {
+						map[i - 1][j] = new MazeTile(i - 1, j, TileType.WALL);
+					}
+
+					//Check East
+					if (i != width - 1 && map[i + 1][j] == null) {
+						map[i + 1][j] = new MazeTile(i + 1, j, TileType.WALL);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Add RoomNode to graph and map.
+	 * Will remove node from the base list of nodes if duplicates are turned off.
+	 * Adds node to usedNodes list.
+	 * Adds entrances to usedEntrancePoints list.
+	 * Adds exits to usedExitsPoints list.
+	 * Adds lights to the lights list.
+	 * 
+	 * Copies the RoomNode if duplicates are allowed.
+	 * 
+	 * @param	node
+	 * 				Node to add
+	 * @param position
+	 * 				position of room
+	 */
+	protected void addRoomNode(RoomNode node, Vec2I position) {
+		if (!duplicates) {
+			baseNodes.remove(node);
+		} else if (node.id != START_ROOM_ID) {
+			node = node.copy(idCounter++);
+		}
+
+		node.carveRoomNode(mazeTiles, position);
+		usedNodes.add(node);
+		usedExitPoints.addAll(node.geOutGoingConnections());
+		usedEntrancePoints.addAll(node.getIncomingConnections());
+		lights.addAll(node.getLights());
+	}
+
+
+	/**
+	 * Initialize all the data containers needed for building.
+	 * @param mapFolder
+	 * 			location of map folder
+	 */
+	protected void initializeBuilder(String mapFolder) {
+		ArrayList<Room> rooms = new ArrayList<Room>();
+		baseNodes = new ArrayList<RoomNode>();
+		usedNodes = new ArrayList<RoomNode>();
+		edges = new HashMap<Integer, CorridorEdge>();
+		usedEntrancePoints = new ArrayList<RoomEntrancePoint>();
+		usedExitPoints = new ArrayList<RoomExitPoint>();
+		try {
+			startAndEndRooms = MapParser.readMap(mapFolder, rooms);
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
Unless we put logging calls back in I think this should not be the only place that logs.